src/components/CustomExtendedDataField/CustomExtendedDataField.js

```diff
@@ -94,15 +94,32 @@ const CustomFieldText = props => {
   );
 };
 
+const validateNumber = (value, max, min, numberTooSmallMessage, numberTooBigMessage) => {
+  const parsedValue = Number.parseInt(value, 10);
+  if (parsedValue > max) {
+    return numberTooBigMessage;
+  }
+  if (parsedValue < min) {
+    return numberTooSmallMessage; 
+  }
+  return undefined; 
+};
+
 const CustomFieldLong = props => {
   const { name, fieldConfig, defaultRequiredMessage, intl } = props;
+  const { maximum, minimum } = fieldConfig;
   const { placeholderMessage, isRequired, requiredMessage } = fieldConfig?.saveConfig || {};
   const label = getLabel(fieldConfig);
-  const validateMaybe = isRequired
-    ? { validate: required(requiredMessage || defaultRequiredMessage) }
-    : {};
   const placeholder =
     placeholderMessage || intl.formatMessage({ id: 'CustomExtendedDataField.placeholderLong' });
+  const numberTooSmallMessage = intl.formatMessage({ id: 'CustomExtendedDataField.numberTooSmall'}, { min: minimum });
+  const numberTooBigMessage = intl.formatMessage({ id: 'CustomExtendedDataField.numberTooBig'}, {max: maximum} );
+
+    const validate = (value, max, min) => {
+      const requiredError = isRequired && !value ? (requiredMessage || defaultRequiredMessage) : undefined;
```

Comment:
If value is 0 this fails.
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Use `value != null` as that checks if value is something else than **null** or **undefined**.